src/api/topup-virtual.api.ts

```diff
@@ -0,0 +1,28 @@
+import { InterceptedAPIHandler } from '../types/base.type';
+import { TopUpVirtualRequest } from '../types/topup-virtual.type';
+
+export const topupVirtual = ({ data, ws, session }: InterceptedAPIHandler) => {
+    const { topup_virtual, req_id } = data as TopUpVirtualRequest;
+    const virtual_account = session.accounts.find(
+        account => account.is_virtual === 1 && account.landing_company_name === 'virtual'
+    );
+
+    const virtual_balance = virtual_account?.balance;
+
+    const amount = 10000 - virtual_balance!;
+
+    const response = {
+        echo_req: { req_id, topup_virtual },
+        msg_type: 'topup_virtual',
+        req_id,
+
+        topup_virtual: {
+            amount,
+            currency: 'USD',
+        },
+    };
+
+    session.accounts.find(account => account.is_virtual === 1 && account.loginid?.startsWith('VRW'))!.balance = 10000;
```

Comment:
```suggestion
    virtual_account.balance = 10000;
```
You should modify the virtual_account directly. It will update the balance